packages/core/src/Platform/index.ts

```diff
@@ -1,30 +1,69 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
+import { CustomUserAgent } from './types';
 import { version } from './version';
+import { detectFramework } from './detectFramework';
+import { UserAgent as AWSUserAgent } from '@aws-sdk/types';
 
-const BASE_USER_AGENT = `aws-amplify/${version}`;
+const BASE_USER_AGENT = `aws-amplify`;
 
 export const Platform = {
-	userAgent: `${BASE_USER_AGENT} js`,
+	userAgent: `${BASE_USER_AGENT}/${version}`,
 	product: '',
 	navigator: null,
 	isReactNative: false,
 };
 if (typeof navigator !== 'undefined' && navigator.product) {
 	Platform.product = navigator.product || '';
 	Platform.navigator = navigator || null;
-	switch (navigator.product) {
-		case 'ReactNative':
-			Platform.userAgent = `${BASE_USER_AGENT} react-native`;
-			Platform.isReactNative = true;
-			break;
-		default:
-			Platform.userAgent = `${BASE_USER_AGENT} js`;
-			Platform.isReactNative = false;
-			break;
+	if (navigator.product === 'ReactNative') {
+		Platform.isReactNative = true;
 	}
 }
 
-export const getAmplifyUserAgent = (content?: string) => {
-	return `${Platform.userAgent}${content ? content : ''}`;
+export const getAmplifyUserAgent = (
+	customUserAgent?: CustomUserAgent
+): AWSUserAgent => {
+	return buildUserAgentTuples(customUserAgent);
+};
+
+export const getAmplifyUserAgentString = (
+	customUserAgent?: CustomUserAgent
+): string => {
+	return buildUserAgentDetailsString(customUserAgent);
+};
+
+const buildUserAgentTuples = (
+	customUserAgent?: CustomUserAgent
+): AWSUserAgent => {
+	const userAgentDetails = {
+		...customUserAgent,
+	};
+	const userAgentTuples: AWSUserAgent = [[BASE_USER_AGENT, version]];
+	if (userAgentDetails.category) {
+		/** TODO: add action as second element */
+		userAgentTuples.push([userAgentDetails.category, undefined]);
+	}
+
+	if (userAgentDetails.framework) {
+		userAgentTuples.push(['framework', userAgentDetails.framework]);
+	} else {
+		userAgentTuples.push(['framework', detectFramework()]);
+	}
```

Comment:
I just noticed a comment of mine got lost into the ether..

I think this can be made more concise without losing clarity:
```suggestion
	userAgentTuples.push(['framework', userAgentDetails.framework ? userAgentDetails.framework : detectFramework()]);
```